src/app/InteractionModelEngine.cpp

```diff
@@ -253,6 +253,8 @@ uint32_t InteractionModelEngine::GetNumActiveWriteHandlers() const
 
 CHIP_ERROR InteractionModelEngine::ShutdownSubscription(const ScopedNodeId & aPeerNodeId, SubscriptionId aSubscriptionId)
 {
+    //ShutdownSubscription should be called from the Matter thread with the lock held
```

Comment:
It can be from some other thread, if it's with the lock held.  I would just remove these comments and leave the asserts: the asserts already pretty clearly document what is being asserted.

Reply:
removed, thanks